### PR TITLE
Back out "[Kineto] Initialize libkineto profilers during torch init process during pybind set-up (#112623)" (#116201)

### PR DESCRIPTION
**Pull request by malfet**
Summary:
This diff needs to be backed out because TorchBench llama_v2_7b_16h has a cublas init error.
https://github.com/pytorch/benchmark/actions/runs/7266269668/job/19797677485?pr=2095

Test Plan: CI

Differential Revision: D52339142

Pull Request resolved: https://github.com/pytorch/pytorch/pull/116201
Approved by: https://github.com/xuzhao9

(cherry picked from commit a357a0f31519f96cff9839c1672a112539ba98ff)

Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by speediedan** [comment]
Thought it might be worth noting that #112623 was also breaking most DDP fork use cases as well (used by downstream packages like Lightning, [Fine-Tuning Scheduler](https://github.com/speediedan/finetuning-scheduler/) etc.). Thanks for making sure it'll be reverted for 2.2.0 (and for all your contributions @malfet!)

**Review by atalman — APPROVED** [review]
